query/src/catalogs/hive/hive_table.rs

```diff
@@ -196,3 +307,68 @@ impl SyncSource for HiveSource {
         Ok(Some(DataBlock::empty_with_schema(self.schema.clone())))
     }
 }
+
+// convert hdfs path format to opendal path formated
+//
+// there are two rules:
+// 1. erase the schema related info from hdfs path, for example, hdfs://namenode:8020/abc/a is converted to /abc/a
+// 2. if the path is dir, append '/' if necessary
+// org.apache.hadoop.fs.Path#Path(String pathString) shows how to parse hdfs path
+pub fn convert_hdfs_path(hdfs_path: &str, is_dir: bool) -> String {
+    let mut start = 0;
+    let slash = hdfs_path.find('/');
+    let colon = hdfs_path.find(':');
+    if let Some(colon) = colon {
+        match slash {
+            Some(slash) => {
+                if colon < slash {
+                    start = colon + 1;
+                }
+            }
+            None => {
+                start = colon + 1;
+            }
+        }
+    }
+
+    let mut path = &hdfs_path[start..];
+    start = 0;
+    if path.starts_with("//") && path.len() > 2 {
+        path = &path[2..];
+        let next_slash = path.find('/');
+        start = match next_slash {
+            Some(slash) => slash,
+            None => path.len(),
+        };
+    }
+    path = &path[start..];
+
+    let end_with_slash = path.ends_with('/');
+    let mut format_path = path.to_string();
+    if is_dir && !end_with_slash {
+        format_path.push('/')
+    }
+    format_path
+}
+
+#[cfg(test)]
```

Comment:
Databend unit test style is not following the Rust in the same file, we separate them to a single folder `tests`.
We can move theis unit test to `tests/it/catalogs` next time, like https://github.com/datafuselabs/databend/tree/main/query/tests/it/catalogs

Reply:
ok,  I'll move the test code to tests in another PR.